src/com/facebook/buck/apple/project_generator/ProjectGenerator.java

```diff
@@ -396,6 +399,10 @@ private boolean shouldGenerateHeaderSymlinkTreesOnly() {
     return options.contains(Option.GENERATE_HEADERS_SYMLINK_TREES_ONLY);
   }
 
+  private boolean shouldGenerateMissingUmbrellaHeaders() {
+    return options.contains(Option.GENERATE_MISSING_UMBRELLA_HEADER);
```

Comment:
My only comment here is that this setting might have to be on a per-target basis. But I think it's okay to cross that bridge when needed as we can just change this method to accept the `TargetNode`.

Reply:
Changing this on a per target basis can be achieved by putting in an empty header with the module name now. We could make this more explicit but I agree that can be left for later.